ckanext/dcat/profiles.py

```diff
@@ -45,6 +47,34 @@
     'owl': OWL,
 }
 
+class CleanedURIRef(object):
+    '''Performs either some basic URL encoding (for http[s]) or whitespace removal on value
+    before creating an URIRef object.
+
+    This is a factory for URIRef objects, which allows usage as type in graph.add()
+    without affecting the resulting node types. That is,
+    g.add(..., URIRef) and g.add(..., CleanedURIRef) will result in the exact same node type.
+    '''
+    @staticmethod
+    def _careful_quote(value):
+        # only encode this limited subset of characters to avoid more complex URL parsing
+        # (e.g. valid ? in query string vs. ? as value).
+        # can be applied multiple times, as encoded %xy is left untouched. Therefore, no
+        # unquote is necessary beforehand.
+        quotechars = ' !"$\'()*,;<>[]{|}'
+        for c in quotechars:
+            value = value.replace(c, quote(c))
+        return value
+
+    def __new__(cls, value):
+        if isinstance(value, basestring):
+            if value.startswith('http'):
+                value = CleanedURIRef._careful_quote(value)
+            else:
+                value = value.replace(' ', '')
```

Comment:
Why do you remove spaces? In the unit test you have the example "mailto:with space@example.com " is converted to "mailto:withspace@example.com". IMHO this is wrong and just buries the error. To remove the spaces from either end of the string (using e.g. `strip()`) could be a solution. But we shouldn't change the data, this should lead to an error and the data should be corrected. Otherwise withspace@example.com might receive emails that were actually meant to with_space@example.com.

Reply:
That sounds sensible to me